### ISSUE TITLE
[BUG] Options not re-rendered when new value is passed to prop

### ISSUE THREAD

**Issue by fmoliveira** [JedWatson/react-select#2744]
Hello,

I know you guys are totally focused on v2 but I've been facing a very annoying bug on v1 and I'm willing to contribute with a fix if I get directions.

## Issue
When my componentDidMount, I load an array into my state with 5 items and it's passed to `options` prop of `<Select />`. When my search function is triggered by `onInputChange` event, my array gets a new value - result of the search, with only 1 item -, and using React Dev Tools I could verify that my new array with only 1 item has been updated in `options` prop. The issue is that the result is not rendered immediately as an `<Option />` inside `Select`; instead, it displays `No results found`.

## Expected Results
When the `option` prop receives a new value it should be updated into the array of `<Option />`'s.

## Workaround
When I clear the search input **or** when I click outside the input to blur it, then click inside it to focus it again, then the result with only 1 array item is shown. But this workaround only work if I interrupt the function called by `onInputChange` to avoid mutating my array again.

## Steps to reproduce
I've been able to reproduce this issue in a sandbox using the latest version of react-select v1, the version 1.2.1. Please go to the link https://codesandbox.io/s/r50v7z32yo and follow these instructions:

### First approach
1. Click in the search input and you'll see 5 initial options
2. Type "admin2", note the message "No results found"
3. Clear the search input
4. Note the result "admin2" is now shown

### Second approach
1. Click in the search input and you'll see 5 initial options
2. Type "admin2", note the message "No results found"
3. Click outside of the search input to blur it
4. Click back inside the search input to focus it
5. Note the result "admin2" is now shown

## Request
If there's already an open issue or PR related to this, please refer me to it so I can get more context. Otherwise, could somebody help me if I need so I could send a PR with a fix? Then could we have a new release with this patch? I'm using react-select in a production environment and this bug is very annoying, it's very important that I get it fixed and I'm willing to work myself on patching it.

Thanks in advance!

**Comment by fmoliveira** [comment]
Anyone @JedWatson @gwyneplaine could please help with directions?

**Comment by fmoliveira** [comment]
My bad @orlov-vo ! It was a typo when sandboxing my issue. I'll verify and come back with an update.

Thank you!

**Comment by fmoliveira** [comment]
Awesome @orlov-vo , thanks to your hint on `filterOption` I have been able to figure out what the issue was on my application.

I use `optionRenderer` to render a full name from an object with first name and last name, but I wasn't aware of the prop `filterOption`. Using it to return the same full name I render solves the issue.

Now I've got a doubt: do I need `labelKey` if I'm using `optionRenderer` and `filterOption`? My render and filter values are computed, hence I don't have a label key.
